### ISSUE TITLE
OpenSSH, with Teleport as jumphost, and session recording "gotcha"

### ISSUE THREAD

**Issue by maznu** [gravitational/teleport#1520]
Is this a bug in Teleport? I think you'll say "no" — and I'd probably agree. Resolution of this issue might be as simple as drawing attention to it (e.g. in documentation) as there is an interplay here which might be an easy way for one of your users to leave a hole in what they believe is a secure Teleport deployment.

In examples below `ssh` is vanilla OpenSSH. I have a `teleportbastion` and a node `box`.

`ssh -J teleportbastion:3022 -p 3022 root@box` records a session. Everything is great, because OpenSSH is talking to Teleport which is proxying through to Teleport.

But if:

1. `box` is running OpenSSH sshd
2. I have a public key on my workstation which is allowed in authorized_keys for root@box (or box's OpenSSH is trusting the Teleport CA, etc).

Then:

`ssh -J teleportbastion:3022 root@box` does not record a session.

At first this was unexpected to me, because I'm jumping through `teleportbastion` — "recording sessions is the whole point of using Teleport!" But after a bit more consideration, and if I understand correctly, the jump from `teleportbastion` is forwarding the (Open)ssh session to `box` over the (Teleport)ssh session with my workstation. In other words: Teleport isn't acting as a "MITM" decrypting proxy, so it doesn't have the unencrypted traffic to be able to store a session.

Risks? I guess a naughty insider could use "-p 3022 -J" to add their (local) public key to root@box, and then using "-J" to do things on `box` which aren't recorded in a session.

I don't have an enterprise license of Teleport to play with, but I surmise that if one were to now add `forward_agent: true` to a role then…

`ssh -t -A -p 3022 teleportbastion ssh root@box`

…will probably record the session (with "do your own risk assessment" caveats about agent forwarding available from, and unencrypted session information flying through, through a hardened host). In some ways, 

tl;dr: mixing OpenSSH and Teleport in a cluster might need to be handled with care.

**Comment by klizhentas** [comment]
Makes sense, I think this feature https://github.com/gravitational/teleport/issues/1510 addresses this exact use case. This is coming in 2.4 release (in a couple of weeks), so stay tuned. Going to close this as fixed via #1510, but feel free to ask any follow up questions in the ticket or create new tickets when convenient!

**Comment by maznu** [comment]
Oh, nice, I hadn't spotted that in the 2.4 roadmap. Excellent — I look forward to it landing :-)
